From 77d26c77b6ac104d3c7b457cd97bee9b4e44128e Mon Sep 17 00:00:00 2001
From: Devajith Valaparambil Sreeramaswamy
 <devajith.valaparambil.sreeramaswamy@cern.ch>
Date: Thu, 31 Oct 2024 09:32:17 +0100
Subject: [PATCH 09/51] Revert "Add RAII to buffer delayed infos, for recursive
 parsing."

This reverts commit d544ff935d05f6a71f1bf7c50d7e3a1b0b415c5d.
---
 .../clang/include/clang/Sema/Sema.h           | 21 -------------------
 1 file changed, 21 deletions(-)

diff --git a/interpreter/llvm-project/clang/include/clang/Sema/Sema.h b/interpreter/llvm-project/clang/include/clang/Sema/Sema.h
index e2b260b904..8404d81bf4 100644
--- a/interpreter/llvm-project/clang/include/clang/Sema/Sema.h
+++ b/interpreter/llvm-project/clang/include/clang/Sema/Sema.h
@@ -1158,27 +1158,6 @@ public:
     }
   };
 
-  class DelayedInfoRAII {
-    Sema &S;
-    SmallVector<std::pair<const CXXMethodDecl*, const CXXMethodDecl*>, 2>
-      DelayedOverridingExceptionSpecChecks;
-    SmallVector<std::pair<FunctionDecl*, FunctionDecl*>, 2>
-      DelayedEquivalentExceptionSpecChecks;
-  public:
-    DelayedInfoRAII(Sema& S): S(S) {
-      std::swap(S.DelayedOverridingExceptionSpecChecks,
-                DelayedOverridingExceptionSpecChecks);
-      std::swap(S.DelayedEquivalentExceptionSpecChecks,
-                DelayedEquivalentExceptionSpecChecks);
-    }
-    ~DelayedInfoRAII() {
-      std::swap(S.DelayedOverridingExceptionSpecChecks,
-                DelayedOverridingExceptionSpecChecks);
-      std::swap(S.DelayedEquivalentExceptionSpecChecks,
-                DelayedEquivalentExceptionSpecChecks);
-    }
-  };
-
   /// WeakUndeclaredIdentifiers - Identifiers contained in \#pragma weak before
   /// declared. Rare. May alias another identifier, declared or undeclared.
   ///
-- 
2.43.5

